src/extractor/edge_based_graph_factory.cpp

```diff
@@ -241,8 +247,7 @@ unsigned EdgeBasedGraphFactory::RenumberEdges()
             // oneway streets always require this self-loop. Other streets only if a u-turn plus
             // traversal
             // of the street takes longer than the loop
-            m_edge_based_node_weights.push_back(edge_data.distance +
-                                                profile_properties.u_turn_penalty);
+            m_edge_based_node_weights.push_back(edge_data.weight);
```

Comment:
@jakepruitt this might actually be a bug due to self-loops in our code. The test forces our path to the right.
When checking for possible shorter paths during contraction, we probably don't consider these `forced` directions. As long as the u-turn is reasonably fast, we could end up omitting a required path.

But this is just a guess here. You could check by (temporarily) adjusting https://github.com/Project-OSRM/osrm-backend/blob/master/include/contractor/graph_contractor.hpp#L824 to always insert edges. If that fixes the problem, we sadly have a bug in relation to bearings and self-loops.